instrumentation/httptrace/clienttrace_test.go

```diff
@@ -134,60 +103,45 @@ func TestHTTPRequestWithClientTrace(t *testing.T) {
 		},
 	}
 	for _, tl := range testLen {
-		span := getSpan(tl.name)
-
+		if !assert.Contains(t, sr, tl.name) {
+			continue
+		}
+		span := sr[tl.name]
 		if tl.parent != "" {
-			parentSpan := getSpan(tl.parent)
-
-			if span.ParentSpanID != parentSpan.SpanContext.SpanID {
-				t.Fatalf("[span %s] does not have expected parent span %s", tl.name, tl.parent)
+			if assert.Contains(t, sr, tl.parent) {
+				assert.Equal(t, span.ParentSpanID(), sr[tl.parent].SpanContext().SpanID)
 			}
 		}
-
-		actualAttrs := make(map[kv.Key]string)
-		for _, attr := range span.Attributes {
-			actualAttrs[attr.Key] = attr.Value.Emit()
-		}
-
-		expectedAttrs := make(map[kv.Key]string)
-		for _, attr := range tl.attributes {
-			expectedAttrs[attr.Key] = attr.Value.Emit()
+		if len(tl.attributes) > 0 {
+			attrs := span.Attributes()
+			// http.local attribute uses a non-deterministic port.
+			delete(attrs, "http.local")
```

Comment:
Maybe `assert.Contains(attrs, "http.local")` prior to deleting it (since `delete` doesn't care whether it exists or not).